### ISSUE TITLE
Sorting Featured Attendees

### ISSUE THREAD

**Issue by willykaram** [NYC-Camp/website#111]
We need the ability to sort/order the attendees displayed on the carousel in the Feature Attendees view. This would be an administrative view on the current carousel view. You would just add it a view block to the current view, that editors can use to set the order of the nodes specifically. We can tackle it however you guys think is best, we just need to be able to allow an non-coder to be able to easily set the order of the nodes in the carousel -- i.e., with draggable view, node queue or queues, etc.. W want to use an approach that is stable and that we can include in our contrib modules that will be added to COD, so we can consider Eric' module, but should keep that point in mind.

**Comment by ForestMars** [comment]
Eh, just add an integer field on those nodes, label it "order," and sort by it in the view, done.

**Comment by willykaram** [comment]
@ForestMars with that approach we have to go in and manually edit all of the nodes individually to adjust the order. @likewhoa @scottrigby and I just dicussed in IRC and Skype too, but using an ordering module is a faster way of organizing the view without needed to edit the nodes individaully to adjust the weights/orders. We also want something that is as useable as possible for contributing back to COD. Adding for discussion purposes: https://drupal.org/node/398508

**Comment by likewhoa** [comment]
I will evaluate both nodequeue and queues module later tonight and come back with a shiny pr.

**Comment by likewhoa** [comment]
After evaluating the available modules for sorting it looks like only queues supports entities but I cannot get it to reference any of the user entities. Nodequeue only works with nodes and it doesn't currently support entities, hence the name. 

I will have to setup a fresh install reproduce the queues issue I am having with user references.

**Comment by scottrigby** [comment]
Hey guys, I'm not sure if I have a different version than what's on GitHub,
or of I just have experience using Queues, or if just works for everyone
else on a clean install, but I just set up a working vanilla D7 User queue
on my laptop on the megabus in 15 mins. I can export it in a feature and
email to someone if that helps?

**Comment by scottrigby** [comment]
OK Fernando, here’s the feature (featured_users.zip).  

You’ll need to make a Queue of the type featured_users (http://d7.kala/admin/content/queues/add/featured_users), and add users to it. I had made a View with a page display listing the added users in a table display. For convenience I also just attached a view with administrative operations to the Page display header (access set to ‘administrator’ role). This should all be easy to tweak, and show it in action.

One thing I noticed is the autocomplete Entityreference field didn’t work properly in the Featured users Queue - that looks like a UI bug, and if so we can fix it later. However, as poor performance as an entity reference select list would be for use by normal users, it works already that way so that’s how I added users (Devel-generated tons of users, and selected some of them in the queue select widget ER field). That bit is slightly annoying till we fix the autocomplete, but I’m guessing this is just for periodic use by an administrator ATM so it shouldn’t hold us up.

Let me  know if this gives you what you need… I’m gonna close my laptop and eyes, but I’ll keep my phone on me - SMS me if it doesn’t make sense =)

**Comment by likewhoa** [comment]
thanks, i'll set this up on my local env first to test. Yea the autocomplete was one of the issues i had.

**Comment by likewhoa** [comment]
http://fr-queues-nyccamp2014.gotpantheon.com is the test environment for this. To manage the queue go to admin/content/queues/manage/2. The view is located at admin/structure/views/view/queues/edit/block

Few issues/bugs
- Any user who has ever changed their username (in this case Andrew Mallis), will need a path alias pointing to their original username url (in this case users/Andrew_Mallis to users/AndrewMallis). Keep in mind that even updating path aliases for users didn't work.
- Any featured attendees must have these fields populated in order for the carousel to show their info. featured_blurb,irc nickname (preferably i would suggest we link irc nickname to their drupal.org user profile page as currently i just have a link to dgo.to/irc/[irc_nickname]),first name & last name

Lets test this carousel on some mobile/tablet devices and once this issue is closed, I will migrate the changes to live.

**Comment by willykaram** [comment]
@likewhoa What is the cause of the path alias issue? It seems odd that we'd be running into this issue. Can you please clarify? Given the issues mentioned above by Scott, if we're running into problems with Queues or NodeQueue can we just implement the Draggable Views as suggested in order to get around this problem?

**Comment by likewhoa** [comment]
The issue is with just one user that I have noticed. If you go to 'users/AndrewMallis' on the live site, you'll notice that his [username] doesn't match that user path alias, so I am assuming Andrew probably changed his [username] at some point for whatever reason. This could possibly be a bug with core since his user patch alias should really be 'users/Andrew_Mallis'.

In terms of using draggableviews module, it will only work with 'table' format last time I tried it and we are not going that route in this case. Queue module has minor issues which shouldn't stop us from using it further. I will explain the few issues here and link to their respected trackers.
- Reference field needs to be saved twice - https://github.com/ericduran/queues/issues/1
- 'Add another' item is funky and doesn't always work - https://github.com/ericduran/queues/issues/8

for the add another item issue, all you have to do is add one queue add a time by saving the whole queue, otherwise it sometimes will wipe the whole queue. For the reference field is just a matter of saving the field twice.

**Comment by willykaram** [comment]
Draggable Views is not limited to working with table formats; the Draggable views table can be used to set the order for any view to which is it associated. Given these bugs, I suggest that we use Draggable Views as we want to be contributing this back to COD, and don't want to be introducing bugs. We can switch the other modules once they are more mature/stable. If others aren't comfortable trying the draggable views route. I can try to get it it later today after completing the sponsorship outreach. It's pretty essential that we get this implemented soon in order to effectively promote the event.

**Comment by likewhoa** [comment]
Ok, I'll look at draggableviews once more, but queues seems like a great module and the current bugs are probably going to be fixed soon.

**Comment by willykaram** [comment]
@likewhoa Can you send me the link to the view you've been working on? I can try to implement draggable views sorting quickly, it should only take 5-10 mins to get this done and we can get it live and focused on other priorities. We can switch the other modules later if they are better once the bugs are resolved.

**Comment by likewhoa** [comment]
it's at http://fr-queues-nyccamp2014.gotpantheon.com/admin/structure/views/view/queues/edit/block

**Comment by willykaram** [comment]
Can you please add the Draggable Views module to the codebase, it's not their yet.
Project: https://drupal.org/project/draggableviews
Instructions: http://www.youtube.com/watch?v=pxetnlKzZBE

**Comment by likewhoa** [comment]
sure, bbiab

**Comment by likewhoa** [comment]
The module should be on the multidev now, have fun :)

**Comment by willykaram** [comment]
I quickly built a new view of users, in which the mater view is the flexslider and the second display is the draggable view to set the order. @likewhoa Can you please tweak the master view to match the prior flexslider that you built and themed, as I though that looked quite good on the front end? Can you also please create a "featured_attendee" image style in image cache which can be used in the feature, so that this is stand alone for purposes of incorporating intro contrib.

Note that the UI for setting the order is accessible here, which allow for easy admin outside of the complexities of the slider/front end of the view:
- http://fr-queues-nyccamp2014.gotpantheon.com/admin/dashboard/featured-attendees

Also note that I used a new featured_attendee field, as the old field has a type on it, with a trailing underscore. Once you're done, we should eliminate that field. I've also created a feature to get this into code, and will comment on that later.

**Comment by willykaram** [comment]
The feature has been added to the mutidev: http://fr-queues-nyccamp2014.gotpantheon.com/admin/structure/features/featured_attendees

**Comment by likewhoa** [comment]
no feature there anymore.

**Comment by likewhoa** [comment]
I will recreate the feature and test locally.

**Comment by likewhoa** [comment]
feature needs work, postpone until tomorrow. Perhaps we can work on it at the playday.

**Comment by ForestMars** [comment]
Setting the order is unnecessary and just creates more work to manage. 

Who decides on the order? Which attendees are more "important"? What's the process for determining that? 

Unless someone has answers for these and related questions that would inevitably come up, this needs to be just a random sort, done.

We need to stop micromanaging every thing which is over taxing our resources and opens us up to issues of consensus and decision making.

**Comment by willykaram** [comment]
@ForestMars This already done/live. The sorting is important for a few reasons, and it's random after the sort, so not much work. This should have been a quick 15 min fix, but we got in a (potentially unnecessary) debate over the approach. We ultimately went with the originally suggested Draggable view module, that was an established bug-free module, and simple approach (it took 15-20 mins to implement start to finish. I agree that down the stretch we need likely have a bit less debate on Down the stretch I think that we need to likely decrease the debate a bit, and focus a bit more on taking the simplest approach and simply getting things done asap.

**Comment by ForestMars** [comment]
> "The sorting is important for a few reasons" 

Which are...?

**Comment by willykaram** [comment]
I don't have time to reply/debate that now ;), but we can do so once time
permits (will reply next week or after the camp). Simply put though, if we
want we only have one shot to highlight both you and webchick or chx or
Alex Pott as speakers in order to attract sponsors/attendees, I'd rather
lead with them (no hard feelings) -- assuming that everyone is not going to
scroll or wait to see other featured attendees). We can largely set it and
forget once the ordering is in place, it took 30 seconds to set the order
we have now, and will take another to do the rest. There a other use cases
planned, but for now this should suffice. It literally took 15-20 mins to
implement (check the IRC logs) and this issue/thread should have ideally
been 2 post: _one and done_. Let's stay focused on the road ahead.
